src/bin/cargo/commands/doc.rs

```diff
@@ -12,6 +12,7 @@ pub fn cli() -> App {
             "open",
             "Opens the docs in a browser after the operation",
         ))
+        .arg(opt("examples", "Document all examples"))
```

Comment:
This doesn't actually do anything; if someone passes the option cargo will ignore it. Consider writing some tests so you can tell whether this works or not, there are lots of examples if you search for  `cargo_test`.

I'm not super familiar with cargo's code, but I expect the right approach would be to loop over all examples and document them (maybe in parallel?). I'd look at how --example x works currently.

Reply:
Thank you for your review, which is a huge encouragement to me. It works fine for my local test.

My test file directory is:
```
hello
├─examples
│    ├───examples1.rs
│    └───examples2.rs
└─src
   └───main.rs
```
![image](https://user-images.githubusercontent.com/46313511/130306822-72c8c1f1-2966-4c9e-9b10-04c1e907a0c9.png)

I think it's good extensibility in the source code.
```
..\cargo\src\cargo\util\command_prelude.rs
```
```
filter: CompileFilter::from_raw_arguments(
                self._is_present("lib"),
                self._values_of("bin"),
                self._is_present("bins"),
                self._values_of("test"),
                self._is_present("tests"),
                self._values_of("example"),
                self._is_present("examples"),     // examples
                self._values_of("bench"),
                self._is_present("benches"),
                self._is_present("all-targets"),
            ),
```